src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -587,14 +588,14 @@ public int deflate(byte[] output, int off, int len, int flush) {
                 inputPos = input.position();
                 int inputRem = Math.max(input.limit() - inputPos, 0);
                 if (input.isDirect()) {
+                    var scope = NIO_ACCESS.acquireSession(input);
                     try {
-                        long inputAddress = ((DirectBuffer) input).address();
                         result = deflateBufferBytes(zsRef.address(),
-                            inputAddress + inputPos, inputRem,
+                                ((DirectBuffer) input).address() + inputPos, inputRem,
```

Comment:
Somewhat subjective but the original code, with inputAddress, was a easier to read and avoided having too much going on in the parameters to deflateBufferBytes. In any case, you probably should fix the formatting to avoid have different parameters with different indentations.